backend/src/packages/chaiNNer_standard/utility/color/split_color.py

```diff
@@ -0,0 +1,102 @@
+from __future__ import annotations
+
+from enum import Enum
+
+from nodes.impl.color.color import Color
+from nodes.properties.inputs import ColorInput, EnumInput
+from nodes.properties.outputs import NumberOutput
+
+from .. import color_group
+
+
+def _norm(n: float) -> int:
+    return int(max(min(n * 255, 255), 0))
+
+
+class SplitColorMode(Enum):
+    NUMBER = 0
+    COLOR = 1
+
+
+output_navi_type = """match Input1 {
+    SplitColorMode::Number => int(0..255),
+    SplitColorMode::Color => Color { channels: 1 },
+    _ => never }
+"""
+
+
+@color_group.register(
+    schema_id="chainner:utility:split_color",
+    name="Split Color",
+    description="Split a color into its RGBA values, either as numbers or grayscale colors.",
+    icon="MdColorLens",
+    inputs=[
+        ColorInput(),
+        EnumInput(SplitColorMode, label="Mode", default=SplitColorMode.NUMBER),
+    ],
+    outputs=[
+        NumberOutput("Red", output_type=output_navi_type),
+        NumberOutput("Green", output_type=output_navi_type),
+        NumberOutput("Blue", output_type=output_navi_type),
+        NumberOutput("Alpha", output_type=output_navi_type),
+    ],
+)
+def split_color_node(
+    color: Color,
+    mode: SplitColorMode,
+) -> tuple[int, int, int, int] | tuple[Color, Color, Color, Color]:
+    length = len(color.value)
+    if length == 1:
+        if mode == SplitColorMode.NUMBER:
```

Comment:
This implementation is overly complex. You essentially need to do 2 things: split the color into RGBA and output the channels. By doing these things after another instead of integrated into each other, the code becomes simpler. So like this:

```py
# get rgba
if len == 1:
    r, g, b, a = ...
elif ...:
    ...

# output
if mode == number:
    return to_num(r), to_num(g), to_num(b), to_num(a)
elif ...:
    ...
```